### PR TITLE
Disable Logs/Metrics on GKE and Run Preemptible Nodes

### PR DESCRIPTION
**Pull request by upodroid**
Part of https://github.com/knative/test-infra/issues/3003

### CODE REVIEW

**Comment by chizhg** [comment]
> Can we make this configurable? For example when debugging a flake I'd like to opt into logging to do better searches

+1 on this.

@upodroid the image with the updated gcloud version has been rolled out, could you make this change?

**Comment by upodroid** [comment]
I added an env variable that enables logging when needed

File: infra-library.sh [hunk 1]
```diff
@@ -137,5 +137,11 @@ function create_gke_test_cluster() {
   local -n _custom_flags=$1
   local -n _test_command=$2
 
-  run_kntest kubetest2 gke "${_custom_flags[@]}" --test-command="${_test_command[*]}"
+# We are disabling logs and metrics on Boskos Clusters by default as they are not used. Manually set GKE_DISABLE_TELEMETRY to false to enable logging
```
Comment (chizhg) [inline infra-library.sh:140]: Super nit: the indentation is not correct.

File: infra-library.sh [hunk 2]
```diff
@@ -137,5 +137,11 @@ function create_gke_test_cluster() {
...
+  if [ "$GKE_DISABLE_TELEMETRY" == "false" ]
```
Comment (chizhg) [inline infra-library.sh:141]: Please keep the Bash style consistent across the scripts to change it to `if [[ "$GKE_DISABLE_TELEMETRY" == "false" ]]; then`.

Also it's more natural to set default value as false, so I would recommend something like `"${GKE_ENABLE_TELEMETRY:=false}"`

**Review by chizhg — COMMENTED** [review]
Left some nits. Also please fix the unit tests.

File: infra-library.sh (hunk 1 above)
Comment (upodroid) [reply infra-library.sh]: fixed

File: infra-library.sh (hunk 2 above)
Comment (upodroid) [reply infra-library.sh]: fixed

File: infra-library.sh [hunk 3]
```diff
@@ -137,5 +137,10 @@ function create_gke_test_cluster() {
   local -n _custom_flags=$1
   local -n _test_command=$2
 
-  run_kntest kubetest2 gke "${_custom_flags[@]}" --test-command="${_test_command[*]}"
+  # We are disabling logs and metrics on Boskos Clusters by default as they are not used. Manually set ENABLE_GKE_TELEMETRY to true to enable logging
+  if [[ "$ENABLE_GKE_TELEMETRY" == "true" ]]; then
+    run_kntest kubetest2 gke "${_custom_flags[@]}" --test-command="${_test_command[*]}" --extra-gcloud-flags="--preemptible"
```
[the remark this answers is not in the document]
Comment (dprotaso) [reply infra-library.sh:142]: cc @chizhg @upodroid

**Pull request merged**